src/realm/util/string_buffer.hpp

```diff
@@ -134,14 +132,16 @@ inline const char* StringBuffer::data() const noexcept
 
 inline char* StringBuffer::c_str() noexcept
 {
+    static char zero = 0;
```

Comment:
My understanding is that the initialization of 'zero' will occur before any checks. But I like the const idea better.